worker/worker.go

```diff
@@ -1480,7 +1480,7 @@ func discardTxnOnErr(ctx context.Context, bus Bus, l *zap.SugaredLogger, txn typ
 
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	if dErr := bus.WalletDiscard(ctx, txn); dErr != nil {
-		l.Errorf("%w: %v, failed to discard txn: %v", *err, errContext, dErr)
+		l.Errorf("%v: %s, failed to discard txn: %v", *err, errContext, dErr)
```

Comment:
What does this fix?

Reply:
This prints something like: `2024-06-03T14:59:53+02:00       ERROR   e2e/cluster_test.go:45  %!w(*errors.errorString=&{foo}): bar context, failed to discard txn: baz` - can't find the exact error so had to quickly reproduce in a test.